datafusion/optimizer/src/simplify_expressions.rs

```diff
@@ -1132,6 +1148,40 @@ mod tests {
         assert_eq!(simplify(expr_plus), lit(2_i64));
     }
 
+    #[test]
+    fn test_simplify_concat_ws_null_separator() {
+        fn build_concat_ws_expr(args: &[Expr]) -> Expr {
+            Expr::ScalarFunction {
+                fun: BuiltinScalarFunction::ConcatWithSeparator,
+                args: args.to_vec(),
+            }
+        }
+
+        let null = Expr::Literal(ScalarValue::Utf8(None));
+        // simple test
+        {
+            let expr = build_concat_ws_expr(&[null.clone(), col("c1"), col("c2")]);
+            assert_eq!(simplify(expr), null);
+        }
+
```

Comment:
I wonder if we could support when arguments other than the first are null? For example

```suggestion
        // simple test
        {
            let expr = build_concat_ws_expr(&[null.clone(), col("c1"), col("c2")]);
            assert_eq!(simplify(expr), null);
        }
        // simple test with null further down the argument list
        {
            let expr = build_concat_ws_expr(&[col("c1"), null.clone(), col("c2")]);
            assert_eq!(simplify(expr), null);
        }
```

I think the current code may only detect NULL in the first argument position